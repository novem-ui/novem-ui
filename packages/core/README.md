# @novem-ui/core

This lib exports all the Novem UI components