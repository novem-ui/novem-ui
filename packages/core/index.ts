export { default as NovemProvider } from './src/novem-provider'

export * from '@novem-ui/badge'

export * from '@novem-ui/avatar'

export * from '@novem-ui/layout'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { default as Checkbox, CheckboxProps } from '@novem-ui/checkbox'

export { Message, MessageProps } from '@novem-ui/message'

export { Box, Flex, BoxProps, FlexProps } from '@novem-ui/layout'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps, Option, OptionProps, Dropdown, DropdownProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'

export * from '@novem-ui/tooltip'
