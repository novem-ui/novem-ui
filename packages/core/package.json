{
  "name": "@novem-ui/core",
  "version": "0.0.1-alpha.1",
  "description": "Core components",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/core#readme",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/core"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/core --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "dependencies": {
    "@novem-ui/badge": "^0.0.1-alpha.1",
    "@novem-ui/theme": "^0.0.1-alpha.1"
  },
  "devDependencies": {
    "@types/novem-theme": "^0.0.1-alpha.1",
    "@types/react": "^17.0.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "polished": "^4.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
