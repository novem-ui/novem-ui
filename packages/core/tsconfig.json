{
  "extends": "../../tsconfig.json",
  "files": [],
  "include": ["**/*.ts", "**/*.tsx", "../../emotion.d.ts"],
  "exclude": ["node_modules/**", "dist/**"]
}
