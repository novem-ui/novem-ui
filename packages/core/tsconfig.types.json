{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.tsx", "**/*.stories.tsx"]
}
