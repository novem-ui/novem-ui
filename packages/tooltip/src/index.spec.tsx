import React, { ReactChild } from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import { NovemProvider, theme } from '@novem-ui/core'

import TooltipBubble, { TooltipBubbleProps } from './tooltip-bubble'
import Tooltip, { TooltipProps } from '.'

const tooltipSetup = (props: TooltipProps = {}) => {
  const result = render(
    <NovemProvider>
      <Tooltip {...props} />
    </NovemProvider>
  )
  return result
}

const bubbleSetup = (props: TooltipBubbleProps & { children?: ReactChild } = {}) => {
  const result = render(
    <NovemProvider>
      <TooltipBubble {...props} />
    </NovemProvider>
  )
  return result
}

describe('components/Tooltip', async () => {
  test('should work on mousover and mouseout', async () => {
    const title = 'Tooltip Title'
    const message = 'message'
    const { container, queryByText } = tooltipSetup({ title, message })

    const tooltipTrigger = container.querySelector("[data-novem='tooltip-trigger']")

    act(() => {
      fireEvent(tooltipTrigger, new MouseEvent('mouseover', { bubbles: true }))
    })

    let tooltipText = queryByText(title)
    let tooltipMessage = queryByText(message)
    expect(tooltipText).toMatchSnapshot()
    expect(tooltipMessage).toMatchSnapshot()

    act(() => {
      fireEvent(tooltipTrigger, new MouseEvent('mouseout', { bubbles: true }))
    })

    tooltipText = queryByText(title)
    tooltipMessage = queryByText(message)

    expect(tooltipText).toBeFalsy()
    expect(tooltipMessage).toBeFalsy()
  })
})

describe('components/TooltipBubble', async () => {
  test('should use the default theme bg', async () => {
    const { container } = bubbleSetup({ position: 'center', children: <p>asd</p>, variant: 'theme' })

    expect(container.firstChild).toHaveStyle({ 'background-color': theme.colors.palette.orange[500] })
    expect(container.firstChild).toMatchSnapshot()
  })

  test('should use the default system color bg', async () => {
    const { container } = bubbleSetup({ position: 'center', children: <p>asd</p>, variant: 'color-scheme' })

    expect(container.firstChild).toHaveStyle({ 'background-color': 'white' })
  })

  test('should be aligned to the center', async () => {
    const { container } = bubbleSetup({ position: 'center' })

    expect(container.firstChild).toHaveStyle({ transform: 'translate(calc(-50% + 1rem), calc(-90%))' })
  })

  test('should be aligned to the left', async () => {
    const { container } = bubbleSetup({ position: 'left' })

    expect(container.firstChild).toHaveStyle({ transform: 'translate(-1rem, calc(-90%))' })
  })

  test('should be aligned to the right', async () => {
    const { container } = bubbleSetup({ position: 'right' })

    expect(container.firstChild).toHaveStyle({ transform: 'translate(calc(-100% + 1.9rem), calc(-90%))' })
  })
})
