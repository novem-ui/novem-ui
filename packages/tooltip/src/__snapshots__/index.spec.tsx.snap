// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Tooltip should work on mousover and mouseout 1`] = `
.emotion-0 {
  font-size: 16px;
  font-weight: bold;
  margin: 0;
}

<p
  class="emotion-0 emotion-1"
>
  Tooltip Title
</p>
`;

exports[`components/Tooltip should work on mousover and mouseout 2`] = `
.emotion-0 {
  font-size: 16px;
  margin: 0;
}

<p
  class="emotion-0 emotion-1"
>
  message
</p>
`;

exports[`components/TooltipBubble should use the default theme bg 1`] = `
.emotion-0 {
  border-radius: 1rem;
  -webkit-filter: drop-shadow(0 1px 3px rgb(0 0 0 / 0.1));
  filter: drop-shadow(0 1px 3px rgb(0 0 0 / 0.1));
  padding: 1rem;
  top: -100%;
  position: absolute;
  width: 11rem;
  -webkit-transform: translate(calc(-50% + 1rem), calc(-90%));
  -moz-transform: translate(calc(-50% + 1rem), calc(-90%));
  -ms-transform: translate(calc(-50% + 1rem), calc(-90%));
  transform: translate(calc(-50% + 1rem), calc(-90%));
  background-color: #F1694B;
}

.emotion-0::after {
  bottom: -0.5rem;
  content: ' ';
  height: 1rem;
  position: absolute;
  -webkit-transform: rotate(45deg) translateY(-2px);
  -moz-transform: rotate(45deg) translateY(-2px);
  -ms-transform: rotate(45deg) translateY(-2px);
  transform: rotate(45deg) translateY(-2px);
  width: 1rem;
  right: 50%;
}

.emotion-0 * {
  color: white!important;
}

.emotion-0::after {
  background-color: #F1694B;
}

<div
  class="emotion-0 emotion-1"
>
  <p>
    asd
  </p>
</div>
`;
